qt/scientific_interfaces/ISISReflectometry/GUI/Batch/BatchPresenter.cpp

```diff
@@ -389,16 +389,38 @@ void BatchPresenter::notifyPreviewApplyRequested() {
   m_experimentPresenter->notifyPreviewApplyRequested(previewRow);
 }
 
-bool BatchPresenter::hasROIDetectorIDsForPreviewRow() const {
+std::map<ROIType, ProcessingInstructions> BatchPresenter::getMatchingProcessingInstructionsForPreviewRow() const {
+  std::map<ROIType, ProcessingInstructions> roiMap;
+
   auto const &previewRow = m_previewPresenter->getPreviewRow();
   try {
-    auto const lookupRow = m_model->findLookupRow(previewRow);
-    if (!lookupRow || !lookupRow->roiDetectorIDs().has_value()) {
-      return false;
+    if (auto const lookupRow = m_model->findLookupRow(previewRow)) {
+      if (auto const signalROI = lookupRow->processingInstructions()) {
+        roiMap[ROIType::Signal] = signalROI.get();
+      }
+      if (auto const backgroundROI = lookupRow->backgroundProcessingInstructions()) {
+        roiMap[ROIType::Background] = backgroundROI.get();
+      }
+      if (auto const transmissionROI = lookupRow->transmissionProcessingInstructions()) {
+        roiMap[ROIType::Transmission] = transmissionROI.get();
+      }
```

Comment:
Could these be refs?

Reply:
Reading around, it looks like you can't use references in a C++ map?

Reply:
Did a bit of local investigating and changing these ones worked but then seemed to cause some issues if you tried to move any of the regions. I think this one can stay as-is